src/dos/drive_fat.cpp

```diff
@@ -1023,22 +1045,26 @@ bool fatDrive::FileCreate(DOS_File **file, char *name, uint16_t attributes) {
 		if(!getDirClustNum(name, &dirClust, true)) return false;
 		memset(&fileEntry, 0, sizeof(direntry));
 		memcpy(&fileEntry.entryname, &pathName[0], 11);
-		fileEntry.attrib = (uint8_t)(attributes & 0xff);
+		fileEntry.attrib  = (uint8_t)(attributes & 0xff);
```

Comment:
Same as above: `static_cast` if sheering is routine and expected or `check_cast` if we expect `attributes` to be in-bounds already.